src/domain/community/user/user.resolver.mutations.ts

```diff
@@ -214,4 +229,47 @@ export class UserResolverMutations {
     );
     return await this.userAuthorizationService.applyAuthorizationPolicy(user);
   }
+
+  private async assignUserToOrganizationByDomain(
+    agentInfo: AgentInfo,
+    user: IUser
+  ): Promise<void> {
+    if (!agentInfo.emailVerified) {
+      return;
+    }
+
+    const userEmailDomain = user.email.split('@')[1];
```

Comment:
it might be a bit of an overkill for this instance, but in my opinion email functions should be under /src/common/utils in something like email.util.ts. There is plenty of email code scattered around the application that can be reused - regexes, splitting, validation etc.